### PR TITLE
Put drops in inventory for survival mode players.

### LINKED ISSUE #889: Survival mode enhancement
When someone is in survival mode and they break a block, instead of the item dropping on the ground (and thus being auto-removed) it would go directly into their inventory. Why this would be useful: ops could set up and play things like Skyblock on the server a lot easier. As it is currently, survival mode play on the server is not practical because of how TFM handles drops, and this would improve that quite a bit without introducing entity lag issues.

### PR DESCRIPTION
**Pull request by LegendIsAwesomes**
Should work in TFM, Resolves #889

### CODE REVIEW

**Comment by JeromSar** [comment]
@LegendIsAwesomes Does not take into account the gamemode of said player.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 1]
```diff
@@ -923,4 +924,17 @@ public void onPlayerLogin(PlayerLoginEvent event)
     {
         TFM_ServerInterface.handlePlayerLogin(event);
     }
+    
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onBreakBlock(BlockBreakEvent event)
+    {
+        final Player player = event.getPlayer();
+        
+        for (ItemStack block : event.getBlock().getDrops())
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: This should probably be `stack` not `block`.

**Comment by LegendIsAwesomes** [comment]
@JeromSar fixed.

**Comment by LegendIsAwesomes** [comment]
@JeromSar, it works, I tested it on immafreedom a server that uses TFM.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 2]
```diff
@@ -32,6 +32,7 @@
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
 import org.bukkit.GameMode;
+import static org.bukkit.GameMode.SURVIVAL;
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Please avoid static imports. :)

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 3]
```diff
@@ -923,4 +925,20 @@ public void onPlayerLogin(PlayerLoginEvent event)
     {
         TFM_ServerInterface.handlePlayerLogin(event);
     }
+    
+    @EventHandler(priority = EventPriority.HIGHEST)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Any reason why this event has a `HIGHEST` priority?

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 4]
```diff
@@ -923,4 +925,20 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+    public void onBreakBlock(BlockBreakEvent event)
+    {
+        final Player player = event.getPlayer();
+        
+        if (player.getGameMode().equals(SURVIVAL))
+        {
+            for (ItemStack stack : event.getBlock().getDrops())
+            {
+                event.getBlock().setType(Material.AIR);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: No need to set the material type to air here. (It's a block break event, this is the default behavior.)

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 5]
```diff
@@ -923,4 +925,20 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+                player.getInventory().addItem(stack);
+                event.setCancelled(true);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Don't cancel the event here.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 6]
```diff
@@ -923,4 +925,20 @@ public void onPlayerLogin(PlayerLoginEvent event)
     {
         TFM_ServerInterface.handlePlayerLogin(event);
     }
+    
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onBreakBlock(BlockBreakEvent event)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: This method is in the wrong class.

**Comment by LegendIsAwesomes** [comment]
Does it go in the server listener?

**Comment by JeromSar** [comment]
@LegendIsAwesomes No the block listener.

**Comment by OnlineRepz** [comment]
@LegendIsAwesomes, any updates?

**Comment by LegendIsAwesomes** [comment]
Will do soon, (Well tomorrow) I have to go for now. I'll leave a reminder for myself.

**Comment by LegendIsAwesomes** [comment]
@JeromSar I didn't do this in the github desktop application so I will have to make a new PR.

**Pull request closed without merging**